src/api/java/com/minecolonies/api/colony/managers/interfaces/IEntityManager.java

```diff
@@ -56,50 +56,50 @@ void sendPackets(
       @NotNull Set<ServerPlayerEntity> newSubscribers);
 
     /**
-     * Returns a map of citizens in the colony. The map has ID as key, and citizen data as value.
+     * Returns a map of civilian in the colony. The map has ID as key, and civilian data as value.
      *
-     * @return Map of citizens in the colony, with as key the citizen ID, and as value the citizen data.
+     * @return Map of civilian in the colony, with as key the civilian ID, and as value the civilian data.
      */
     @NotNull
-    Map<Integer, ICitizenData> getCitizenMap();
+    Map<Integer, ICivilianData> getCivilianDataMap();
 
     /**
-     * Get citizen by ID.
+     * Get civilian by ID.
      *
-     * @param citizenId ID of the Citizen.
-     * @return ICitizenData associated with the ID, or null if it was not found.
+     * @param civilianId ID of the civilian.
+     * @return ICivilianData associated with the ID, or null if it was not found.
      */
-    ICitizenData getCitizen(int citizenId);
+    <T extends ICivilianData> T getCivilian(int civilianId);
 
     /**
-     * Spawns a citizen with the specific citizen data.
+     * Spawns a civilian with the specific civilian data.
      *
      * @param data     Data to use when spawn, null when new generation.
      * @param world    THe world.
-     * @param force    True to skip max citizen test, false when not.
      * @param spawnPos the pos to spawn it at.
-     * @return the new citizen.
+     * @param force    True to skip max civilian test, false when not.
+     * @return the new civilian.
      */
-    ICitizenData spawnOrCreateCitizen(ICitizenData data, World world, BlockPos spawnPos, boolean force);
+    <T extends ICivilianData> T spawnOrCreateCivilian(ICivilianData data, World world, BlockPos spawnPos, boolean force);
```

Comment:
You probably also want to take in the type T instead of ICivilianData as the first parameter?

Reply:
ye